src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,238 @@
+            String by = deadlineDescription[1].trim();
+    
+            Deadline newDeadline = new Deadline(description, by);
+            this.taskList.add(newDeadline);
+               
+            System.out.println("\t" + "Added deadline: " + newDeadline.toString());
+            storage.saveTaskListToFile();
+        }
+    }
+
+    /** 
+     * Adds eventTask to taskList
+     * 
+     * @param task task to be added
+     */
+    public void addEvent(String task) {
+    
+        String[] eventParts = task.substring(6).trim().split("/from");
+        if (eventParts.length == 2) {
+            String[] durationParts = eventParts[1].trim().split("/to");
+            if (durationParts.length == 2) {
+                String desc = eventParts[0].trim();
+                String from = durationParts[0].trim();
+                String to = durationParts[1].trim();
+                Events newEvent = new Events(desc, from, to);
+                this.taskList.add(newEvent);
+                   
+                System.out.println("\t" + "Added event: " + desc + " (from: " + from + ", to: " + to + ")");
+                storage.saveTaskListToFile();
+    
+            } else {
+                System.out.println("\t" + "Invalid input for event. Please use the format: event <task> /from <start time> /to <end time>");
+            }
+        } else {
+            System.out.println("\t" + "Invalid input for event. Please use the format: event <task> /from <start time> /to <end time>");
+        }
+    }
+
+    /** 
+     * Deletes task from taskList
+     * 
+     * @param index index of task to be deleted
+     * @throws IndexOutOfBoundsException if index is out of bounds
+     * @throws NumberFormatException if input is not a number
+     */
+    public void deleteTask(int index) {
+        try {
+            System.out.println("\t" + "Noted. I've removed this task:" + "\n" + "\t" 
+                    + "[ " + this.taskList.get(index) + " ]");
+            this.taskList.remove(index);
+            System.out.println("\t" + "There are " + this.taskList.size() + " tasks in your list.");
+            storage.saveTaskListToFile();
+        } catch (IndexOutOfBoundsException e) {
+            ui.printInvalidTaskIndex();
+        } catch (NumberFormatException e) {
+            ui.printInvalidTaskIndex();
+        }
+    }
+
+    /**
+     * Returns task type
+     * 
+     * @param task task to be added
+     * @return TaskType
+     */
+    public TaskType getTaskType(String task) {
+        if (task.startsWith("todo")) {
+            return TaskType.TODO;
+        } else if (task.startsWith("deadline")) {
+            return TaskType.DEADLINE;
+        } else if (task.startsWith("event")) {
+            return TaskType.EVENT;
+        } else if (task.equals("help")) {
+            return TaskType.HELP;
+        } else if (task.startsWith("delete")) {
+            return TaskType.DELETE;
+        } else {
+            return TaskType.UNKNOWN;
```

Comment:
Was there a reason you chose to explicitly include an `UNKNOWN` task type? I thought the `default` part of your `switch` statement would have already caught any invalid commands.

Reply:
youre right thats an oversight on my part